src/main/java/Task.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Encapsulates a task which other specific tasks extend from.
+ */
+public class Task {
+
+    /**
+     * Member variables holding name and status of task.
+     */
+    private String name;
+    private boolean isDone;
+
+    /**
+     * Constructor to create a task.
+     *
+     * @param name Name of task.
+     */
+    public Task(String name) {
+        this.name = name;
+        this.isDone = false;
+    }
```

Comment:
Consider only using of <code>this.</code> when referencing attributes that have been shadowed by method/constructor parameter names